kernel/Utilities/SparsityEstimator.cpp

```diff
@@ -15,310 +18,416 @@ SparsityEstimator::SparsityEstimator(const GlobalIndexing& rowindexing,
                                      const GlobalIndexing& columnindexing)
     : rowIndexing_(rowindexing), columnIndexing_(columnindexing) {}
 
-struct SparsityEstimator::Workspace {
-    /// DoFs owned by the current processor. Stored as (dof, size) where dof is
-    /// the first dof of an element, face, etc.
-    std::map<std::size_t, std::size_t> owned;
-    // Same as #owned, but for the DoFs not owned by the current processor
-    std::map<std::size_t, std::size_t> notOwned;
+namespace Detail {
+/// An interval (start, size) of global DoF-Indices
+/// We assume that each interval is for a single unknown and all the DoFs in the
+/// interval are associated with the same Element, Face, Edge or Node.
+using DoFInterval = std::pair<std::size_t, std::size_t>;
 
-    /// Get a reference to either the owned or notOwned DoF set.
-    std::map<std::size_t, std::size_t>& getDoFs(bool owning) {
-        return owning ? owned : notOwned;
-    }
+/// Set of DoFIntervals
+using DoFIntervals = std::set<DoFInterval>;
+
+/// Enum to specify rows or columns of a matrix
+enum MatrixSide { ROWS, COLUMNS };
```

Comment:
Hmm, that would give me the impression that we are talking about how something is ordered. The word side is not really nice, because that could also mean left or right hand side of the matrix.

Here I'm just wanting to know whether a structure is describing the rows or the columns of a matrix. Thinking of it `RowsOrColumns` sounds a bit long/strange. So have to think about it.

Reply:
`Indexing` ?

Reply:
Indexing is already used for something else (for each row/column to which degree of freedom does it correspond).

I've changed it to `RowsOrColumns` find that rather direct and it does not take that much more space.